cloud_Azure/terraform/module/examples/multiple_accounts_multiple_resource_group/profiles_populator.py

```diff
@@ -0,0 +1,428 @@
+        config[section]["resource_group_names"] = ",".join(profile.resource_group_names)
+        config[section]["storage_account_names"] = ""
+
+    try:
+        with open(file_path, "w", encoding="utf-8") as configfile:
+            config.write(configfile)
+        log.info("Saved profiles to '%s'", file_path)
+        return True
+    except OSError:
+        log.exception("Failed to save profiles to '%s'", file_path)
+        return False
+
+
+def profile_exists(profile_name: str, profiles: List[AzureProfile]) -> bool:
+    return find_profile(profile_name, profiles) is not None
+
+
+def find_profile(profile_name: str, profiles: List[AzureProfile]) -> Optional[int]:
+    for i, profile in enumerate(profiles):
+        if profile.name == profile_name:
+            return i
+    return None
+
+
+def find_secret(principal_id: str, profiles: List[AzureProfile]) -> Optional[str]:
+    for profile in profiles:
+        if profile.principal_id == principal_id and likely_valid_service_principal_secret(profile.principal_secret):
+            log.info("Secret for Service Principal ID '%s' found in profile '%s'", principal_id, profile.name)
+            return profile.principal_secret
+    log.info("Secret for Service Principal ID '%s' not found", principal_id)
+    return None
+
+
+def likely_valid_service_principal_secret(secret: str) -> bool:
+    SERVICE_PRINCIPAL_SECRET_LENGTH = 34
+    return len(secret) == SERVICE_PRINCIPAL_SECRET_LENGTH
+
+
+def insert_or_overwrite_profile(profiles: List[AzureProfile], profile: AzureProfile) -> None:
+    index = find_profile(profile.name, profiles)
+    if index is not None:
+        profiles[index] = profile
+    else:
+        profiles.append(profile)
+
+
+def backup_file(file_path: str) -> bool:
+    """Create file backup with time stamp suffix in the name"""
+
+    if not os.path.exists(file_path):
+        log.info("Source file '%s' doesn't exist. Skipping backup", file_path)
+        return True  # nothing to backup -> OK
+
+    backup_time = datetime.today().strftime("%Y-%m-%d_%H:%M:%S")
+    backup_name = f"{file_path}.{backup_time}"
+    backup_dir = Path(BACKUP_PROFILES_DIRECTORY)
+    backup_path = backup_dir.joinpath(backup_name)
+    try:
+        backup_dir.mkdir(exist_ok=True)
+        shutil.copyfile(file_path, backup_path)
+    except OSError:
+        log.exception("Failed to create backup '%s' of '%s'", backup_path, file_path)
+        return False
+
+    log.info("Created backup '%s' of '%s", backup_path, file_path)
+    return True
+
+
+def print_log_in_columns(items: List[Any], num_columns: int) -> None:
+    it = iter(items)
+    groups = iter(lambda: tuple(islice(it, num_columns)), ())
+
+    format_str = "{: <20} " * num_columns
+    for row in groups:
+        print_log(format_str.format(*row))
+
+
+def parse_cmd_line() -> str:
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--filename", default=DEFAULT_PROFILES_FILE_NAME, help="Profiles file name")
```

Comment:
We should have an option to enable debug messages.

Reply:
You mean: to specify logging level?

Reply:
I think that for simple utility like this, 2 log levels are sufficient:
- normal (enabled by default)
- debug/verbose

So, all we need is to enable debug/verbose logging.

Reply:
Added "--verbose" cmd line flag